### DISCUSSION TITLE
List of text editors support Modula-3

### DISCUSSION THREAD

**Discussion by ghost** [modula3/cm3#1130] [General]
I know KDE's Kate and JEdit support Modula-3 syntax highlighting. KDE's Kate offers some basic code completion based on tags file I think.

Update: since version 24.3.0, Text Editor Pro supports syntax highlighting for Modula-3: https://texteditor.pro/index.php?page=history

**Comment by mikanystrom** [comment]
I use the modula3.el package in emacs.  It does indentation and keyword completion well, only a few minor bugs.  Doesn't do anything else though.

**Comment by RodneyBates** [comment]
On 12/29/22 01:47, tn1997tn wrote:
>
> I know KDE's Kate and JEdit support Modula-3 syntax highlighting. KDE's Kate offers some basic code completion based on tags file I think.
>

Schutz does Modula-3 syntax highlighting, or used to.  The display of
colors and faces has somehow gotten undermined by changes in the
fonts available in newer distributions, that I haven't had time to
figure out.  Temporarily, I am, using background colors instead--
rather poor-looking IMO, but it does show the distinctions.


Schutz makes suggested corrections to syntactically incorrect code.
Not very good ones, but you can accept them and then reparse.  I
hadn't thought about it, but it might serve as completion, for
certain construct prefixes.  A simpler user interface would help.

**Reply by ghost** [reply]
What is Schutz?

**Comment by RodneyBates** [comment]
On 12/31/22 01:54, tn1997tn wrote:
>
> What is Schutz?
>

Go to github.com/RodneyBates/schutz and begin by looking in
doc/pdf.

**Comment by nsgnkhibdk2cls0f** [comment]
This text editor also supports Modula-3: https://github.com/stefankueng/BowPad

**Comment by MarcusE1W** [comment]
I have found this the other day (not from me)

A new lexer for the textadept editor

https://github.com/paaguti/textadept-modula-lexers
